### ISSUE TITLE
closePanel got stuck sometimes

### ISSUE THREAD

**Issue by xlongtang** [nolimits4web/Framework7#940]
Hi, there is an issue about closing cover panels ...See below for details 

With a right cover effect panel, the behavior of closing panels was wrong sometimes. Specifically, the part of code on 'TransitionEnd' did not get executed. 
Therefore, the app ends with a state with the following characteristics: 
- The body element has a class of "panel-closing"
- The panel overlay is visible 
- The cover panel is visible with 'display:block'
- the UI is not responsive anymore (obvious)

The above scenarios occurred in Android systems...

Please investigate this issue...
Thanks.
Best, 
Xiaolong

**Comment by nolimits4web** [comment]
Please provide some actions/instruction to replicate this issue. It is hard to replicate "sometimes"

**Comment by WouterBothoff** [comment]
Same problem here with right panel. On the iPhone when you use swipe to open the panel about 25%, then hold it with your finger and double click on a menu item. The app goes to the new page and the view is stuck, the panel stays open and the view doesn't respond to clicks at all. Sometimes when you swipe and click to fast it also gets stuck!

**Comment by wolfroma** [comment]
I have the same problem. After debugging I found that `closePanel` doesn't set `app.allowPanelOpen = true;`. All consequent `closePanel` requests have no effect because the condition:

`app.closePanel = function () {`
`            var activePanel = $('.panel.active');`
`            if (activePanel.length === 0) return false;`
`           ...`
`}`


So on panel open it exits right away because of the condition in `openPanel` method
`app.openPanel = function (panelPosition) {`
`            if (!app.allowPanelOpen) return false;`
`}`

It seems  the logic of the `allowPanelOpen` flag is fragile. .
Adding extra condition helped me, but I'm not sure if it defeats the purpose of the allowPanelOpen flag:
`app.openPanel = function (panelPosition) {`
`            if (!app.allowPanelOpen && $('.panel.active').length !== 0 ) return false;`

**Comment by wolfroma** [comment]
After some time of debugging I found that mistake in app logic caused such behavior. Eventually all classes were removed from the body tag. Since the framework keeps record of panel states there, it broke the framework logic. Even if I didn't use the framework properly I think it would be great to add extra guard checks somewhere in order to make sure that the framework can survive or give a warning in such conditions.
